File: lab/javascript/matematica_e_operadores/maior_e_menor_em_lista.js
const prompt = require('prompt-sync')();

function maior_e_menor_em_uma_lista(){
    let limite = Number(prompt());

    let entrada = prompt();
    let lista = entrada.split(" ").map(Number);

    if (lista.length > limite || lista.length < limite){
        throw new Error(
            "Tamanho da lista precisa compadecer ao limite esperado"
        );
    }

    let maior = lista[0];
    let menor = lista[0];

    lista.forEach(element => {
        if (element > maior){
            maior = element;
        }

        if (element < menor){
            menor = element;
        }
    });

    console.log("Menor:", menor);
    console.log("Maior:", maior);
}

function maior_e_menor_em_uma_lista2(){
    let limite = Number(prompt());

    let entrada = prompt();
    let lista = entrada.split(" ").map(Number);

    if (lista.length > limite || lista.length < limite){
        throw new Error(
            "Tamanho da lista precisa compadecer ao limite esperado"
        );
    }

    console.log("Menor:", Math.min(...lista));
    console.log("Maior:", Math.max(...lista));
}

maior_e_menor_em_uma_lista2();

File: lab/python/matematica_e_operadores/maior_e_menor_em_lista.py
def maior_e_menor_em_uma_lista():
    limite = int(input())
    lista = list(map(int, input().split()))

    if len(lista) > limite or len(lista) < limite:
        raise IndexError(
            "Tamanho da lista precisa compadecer ao limite esperado"
        )

    maior = int(lista[0])
    menor = int(lista[0])
    for i in lista:
        numero = int(i)
        if numero > maior:
            maior = numero

        if numero < menor:
            menor = numero

    print("Menor:", menor)
    print("Maior:", maior)


def maior_e_menor_em_uma_lista2():
    limite = int(input())
    lista = list(map(int, input().split()))

    if len(lista) > limite or len(lista) < limite:
        raise IndexError(
            "Tamanho da lista precisa compadecer ao limite esperado"
        )

    print("Menor:", min(lista))
    print("Maior:", max(lista))


maior_e_menor_em_uma_lista2()
